trezorlib/tests/device_tests/test_msg_cardano_sign_transaction.py

```diff
@@ -0,0 +1,137 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2018 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+from binascii import unhexlify, hexlify
+import pytest
+import time
+import binascii
+
+from .common import TrezorTest
+from trezorlib import messages
+from trezorlib.cardano import cardano_sign_transaction, create_sign_transaction_message, create_tx_ack_message
+
+
+@pytest.mark.cardano
+@pytest.mark.skip_t1  # T1 support is not planned
+class TestMsgCardanoSignTransaction(TrezorTest):
+
+    def test_cardano_sign_transaction(self):
+        self.setup_mnemonic_nopin_nopassphrase()
```

Comment:
ISTM this test could be vastly simplified if you used `set_expected_responses`

see e.g. https://github.com/trezor/python-trezor/blob/master/trezorlib/tests/device_tests/test_msg_signtx.py

Reply:
I want to used that but I was unsure how to use it when I need to do swipes

Reply:
Ah, the swipes thing again! Yes, it makes sense in that case. We need a better infrastructure for simulating swipes.